packages/edit-post/src/components/start-page-options/index.js

```diff
@@ -63,70 +62,55 @@ function PatternSelection( { onChoosePattern } ) {
 	);
 }
 
-const START_PAGE_MODAL_STATES = {
-	INITIAL: 'INITIAL',
-	PATTERN: 'PATTERN',
-	CLOSED: 'CLOSED',
-};
-
-export default function StartPageOptions() {
-	const [ modalState, setModalState ] = useState(
-		START_PAGE_MODAL_STATES.INITIAL
-	);
-	const blockPatterns = useStartPatterns();
-	const hasStartPattern = blockPatterns.length > 0;
-	const shouldOpenModel = useSelect(
-		( select ) => {
-			if (
-				! hasStartPattern ||
-				modalState !== START_PAGE_MODAL_STATES.INITIAL
-			) {
-				return false;
-			}
-			const { getEditedPostContent, isEditedPostSaveable } =
-				select( editorStore );
-			const { isEditingTemplate, isFeatureActive } =
-				select( editPostStore );
-			return (
-				! isEditedPostSaveable() &&
-				'' === getEditedPostContent() &&
-				! isEditingTemplate() &&
-				! isFeatureActive( 'welcomeGuide' )
-			);
-		},
-		[ modalState, hasStartPattern ]
-	);
+function StartPageOptionsModal() {
```

Comment:
I believe there's also StartTemplateOptions in the site editor, I wonder if it suffers the same issue.

Reply:
That's a good point, I reviewed `StartTemplateOptions`, and found that it also parses patterns, but it already uses the "two nested components" technique that my PR introduces for `StartPageOptions`. The outer component checks the pre-conditions, and only the inner component starts working with patterns.